scene/gui/graph_edit.cpp

```diff
@@ -1622,6 +1904,16 @@ void GraphEdit::_update_zoom_label() {
 	zoom_label->set_text(zoom_text);
 }
 
+void GraphEdit::_invalidate_connection_line_cache() {
+	for (Ref<Connection> &c : connections) {
+		c->_cache.dirty = true;
+	}
+}
+
+float GraphEdit::_get_shader_line_width() {
+	return MAX(0, lines_thickness * theme_cache.base_scale + 4.0);
```

Comment:
Is there a case when thickness or scale can be negative? Otherwise, `MAX` is not really needed, as `+4` guarantees positivity.

Reply:
Well, the inspector currently allows setting negative values which makes no sense, so I just added a property hint: `PROPERTY_HINT_RANGE, "0,100,0.1,suffix:px"` and a check to the setter
`ERR_FAIL_COND_MSG(p_thickness < 0, "Connection lines thickness must be greater than or equal to 0.");`.
Now `MAX` is definitely not needed anymore.

Reply:
Uhm, that's what I meant with :)
> and a check to the setter
ERR_FAIL_COND_MSG(p_thickness < 0, "Connection lines thickness must be greater than or equal to 0.");

Reply:
Sorry, I'm blind <3 All good!